### PR TITLE
lvm-operator: add initial CI config

### PR DESCRIPTION
**Pull request by sp98**
Signed-off-by: Santosh Pillai <sapillai@redhat.com>

### CODE REVIEW

File: ci-operator/config/red-hat-storage/lvm-operator/OWNERS [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# approval == this is a good idea /approve
+approvers:
+  - nbalach
+# review == this code is good /lgtm
+reviewers:
+  - nbalacha
+  - rohantmp
+  - leelavg
+  - sp98
+  - jmolmo
```
Comment (nbalacha) [inline ci-operator/config/red-hat-storage/lvm-operator/OWNERS:10]: You can remove jmolmo for now as he is unlikely to have time for this project.

File: ci-operator/jobs/red-hat-storage/lvm-operator/OWNERS [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# approval == this is a good idea /approve
+approvers:
+  - nbalach
```
Comment (nbalacha) [inline ci-operator/jobs/red-hat-storage/lvm-operator/OWNERS:3]: typo: nbalacha

File: ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-main.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      name: "4.10"
+      namespace: ocp
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: unit-test
+  commands: go test ./... -tags unit -mod=readonly -coverprofile cover.out
```
Comment (nbalacha) [inline ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-main.yaml:47]: would `make test` work here?

Comment (sp98) [reply ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-main.yaml:47]: Yes. I'm testing a few things out.  So its not really ready for review yet.

File: ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-release-4.10.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+base_images:
+  base:
+    name: "4.10"
+    namespace: ocp
+    tag: base
+  opm-builder:
+    name: opm-builder
+    namespace: ci
+    tag: v1.14.3
+binary_build_commands: go mod vendor make build build-vgmanager
```
Comment (nbalacha) [inline ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-release-4.10.yaml:10]: Why use separate binaries here?

File: ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-main.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+test_binary_build_commands: |
+  go mod vendor
+  mkdir /.cache && chmod 775 -R /.cache ${GOPATH}
```
Comment (sp98) [inline ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-main.yaml:29]: @jarrpa I think you might have face similar permission issues. How did you manage to work around permission issues to create ./cache dir while working on the CI?

Comment (jarrpa) [reply ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-main.yaml:29]: Best I know we don't:
* CI config: https://github.com/openshift/release/blob/master/ci-operator/config/red-hat-storage/ocs-operator/red-hat-storage-ocs-operator-main.yaml#L82
* Actual tests: https://github.com/red-hat-storage/ocs-operator/blob/main/Makefile#L164

File: ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-main.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+binary_build_commands: make build build-vgmanager
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.16
```
Comment (mykaul) [inline ci-operator/config/red-hat-storage/lvm-operator/red-hat-storage-lvm-operator-main.yaml:6]: OCP 4.10 and 4.11 use go 1.17.5 (changed from 1.17.2) that solve important CVEs

**Review by nbalacha — COMMENTED** [review]
/lgtm

**Review by agarwal-mudit — APPROVED** [review]

**Pull request merged**